### ISSUE TITLE
image classification benchmark and finetune

### ISSUE THREAD

**Issue by jacquesqiao** [PaddlePaddle/Paddle#5892]
project: https://github.com/PaddlePaddle/Paddle/issues/6024

**Comment by jacquesqiao** [comment]
### Paddle fluid:
Total examples: 1120, total time: 162.26692
6.90221 examples/sec, 5.07084 sec/batch
pass_acc: [ 0.07985038]

### tf
Total examples: 1120, total time: 69.05409
16.21917 examples/sec, 2.15794 sec/batch

**Comment by jacquesqiao** [comment]
### Tensorflow
pass=3, batch=192, loss=1.980749, acc=0.400000

Total examples: 1120, total time: 236.70867
4.73155 examples/sec, 7.39715 sec/batch

**Comment by qingqing01** [comment]
@jacquesqiao I tested PaddlePaddel fluid in TITAN X (Pascal) by using cuDNN and remove the [stream synchronization between each operator](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/operator.cc#L431). The total time of one pass of PaddlePaddle fluid is 70.47459 sec. From your testing, the time of TensorFlow is 69.05409 sec.  

Config:
   - 1 pass
   - batch_size: 32
   - Singe GPU
   - cuDNN 5.1
  
Time:

Total examples: 6149, total time: 70.47459
87.25131 examples/sec, 0.36515 sec/batch

**Comment by jacquesqiao** [comment]
@qingqing01 cool!, I will also have a try~

**Comment by dzhwinter** [comment]
If we use the `Finish` implement the synchronized training. That's really hurt performance a lot when doing the training.  How about put `Finish` barrier at the executor finish line?

**Comment by jacquesqiao** [comment]
### condition: 
- use a fixed numpy as data.
- drop the first 10 iteration

### result:
- Fluid:
Total examples: 2240, total time: 16.97440
131.96342 examples/sec, 0.21218 sec/batch

- Tensorflow:
Total examples: 2240, total time: 15.94563
140.47733 examples/sec, 0.49830 sec/batch
